dashboard/test_dash_app.py

```diff
@@ -77,3 +83,122 @@ def test_handle_logout(mock_session_state):
     logout_of_dashboard(cookie_manager)
 
     assert mock_session_state.__setitem__.call_count == 3
+
+############ database.py ############
+
+
+@patch.dict("os.environ", {
+    "DB_USER": "test_user",
+    "DB_PASSWORD": "test_password",
+    "DB_HOST": "test_host",
+    "DB_PORT": "1234",
+    "DB_NAME": "test_db"})
+@patch("database.connect")
+def test_get_database_connection_successful(mock_connect):
+    """
+    Testing connection to database
+    """
+    expected_connection = MagicMock()
+    mock_connect.return_value = expected_connection
+    result = get_database_connection()
+
+    assert result == expected_connection
+
+
+@patch.dict("os.environ", {
+    "DB_USER": "test_user",
+    "DB_PASSWORD": "test_password",
+    "DB_HOST": "test_host",
+    "DB_PORT": "1234",
+    "DB_NAME": "test_db"})
+@patch("database.connect", side_effect=ConnectionError("Connection error"))
+def test_get_database_connection_connection_error(mock_connect):
+    """
+    Testing failed connection to database
+    """
+    result = get_database_connection()
+
+    assert isinstance(result, ConnectionError)
+
+
+@patch("database.connect")
+def test_load_all_database_info(mock_connect):
+    """
+    Test that checks data is returned in the format of COLUMNS as specified in the SELECT_ALL_QUERY
+    """
```

Comment:
Could put a full stop at the end of the docstring just to make it consistent with all other docstrings in this repo.